### DISCUSSION TITLE
TypeError: Invalid attempt to spread non-iterable instance.

### DISCUSSION THREAD

**Discussion by cargallo** [react-hook-form/react-hook-form#5831] [Q&A]
I know this doesn't follow the issues template but still want to report de bug until I can reproduce it in a sandbox.
I think its related to the latest changes in 7.10.0 related to this PR https://github.com/react-hook-form/react-hook-form/pull/5705

**(UPDATE INFO: searching into github changes in the files mentioned I decided to downgrade to version 7.8.6 and it got solved so its related to that change I mentioned or its predecessors)**

It happens when I try to append into fieldArray
```JSX
  45 |  const handleAgregarDireccion = () => {
> 46 |    domiciliosAppend({ direccion: null });
     | ^  47 |  }
```

TypeError: Invalid attempt to spread non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.

src/utils/append.ts:4
```JSX
  1 | import convertToArrayPayload from './convertToArrayPayload';
  2 | 
  3 | export default function append<T>(data: T[], value: T | T[]): T[] {
> 4 |   return [...data, ...convertToArrayPayload(value)];
  5 | }
  6 | 
```
src/useFieldArray.ts:138
```JSX
  135 | ) => {
  136 |   inFieldArrayActionRef.current = true;
  137 |   if (get(fieldsRef.current, name)) {
> 138 |     const output = method(get(fieldsRef.current, name), args.argA, args.argB);
      | ^  139 |     shouldSet && set(fieldsRef.current, name, output);
  140 |   }
  141 | 
```

src/useFieldArray.ts:246
```JSX
 243 |     FieldArray<TFieldValues, TFieldArrayName>
  244 |   >[],
  245 | );
> 246 | batchStateUpdate(
      | ^  247 |   appendAt,
  248 |   {
  249 |     argA: fillEmptyArray(value),
```

**Comment by bluebill1049** [comment]
thanks for reporting the issue @cargallo. can you provide me with a CSB so I can look into the reason behind it? It could be usage issue or bug within the lib.

**Reply by cargallo** [reply]
I tried but I cant reproduce it. Its really difficult because I think its related with some kind of shallow copy, but believe me it's not my code. 

Besides that fieldArray is working strange, setValue is not updating fields of fieldArray and some stranges behaviours.

I know you did that to improve performance but in the code even in your examples you recommend this way of use:

```JSX
const {fields} = useArrayField();

<div>
{fields.map((item, index)=>....)}
</div>
```
So if I want to update a value in fieldArray via setValue. That doesn't impact in the view. I can only remove, insert, append.

**Reply by bluebill1049** [reply]
> setValue is not updating fields of fieldArray and some stranges behaviours.

can you share a codesandbox?

**Reply by bluebill1049** [reply]
> I tried but I cant reproduce it. Its really difficult because I think its related with some kind of shallow copy, but believe me it's not my code.

I do. Need to reproduce the issue, because code don't lie, that commit that you pointed out, doesn't looks like the root cause of the issue either (by description).

**Reply by saschagehlich** [reply]
I'm also experiencing this, also with 7.8.6. Unfortunately I don't have time to create a CSB right now

**Reply by saschagehlich** [reply]
@bluebill1049 Here's an example reproducing the issue. Just hit the "add items" button: https://codesandbox.io/s/vigorous-sara-m5hg6

EDIT: It only happens when `register()`ing the field

**Reply by revich2** [reply]
@saschagehlich @cargallo I had this issue, but after some investigation i found problem in my code. in one of useEffects i used form.reset with undefined value and this problem happened. in investigation process i found some strange things, which solves this, but affects my functionality. try check your code, because i also tried reproduce in CSB, but i couldn't do it.
